p2p/protocols/protocol.go

```diff
@@ -274,6 +288,15 @@ func (p *Peer) Send(ctx context.Context, msg interface{}) error {
 		Payload: r,
 	}
 
+	//if the accounting hook is set, call it
+	if p.spec.Hook != nil {
+		err := p.spec.Hook.Send(p, wmsg.Size, msg)
+		if err != nil {
+			p.Drop(err)
```

Comment:
I don't think this is responsibility of the `p2p/protocol` package.

Reply:
indeed, returning the error in the `Send` will result in a peer drop further up

Reply:
nope. Its the other way  round. 
here it is part of the contract with the hook: if presend hook call errors we break the connection. 
if you dont want this semantics, dont return an error in your hook